### PR TITLE
Filter online offline workers

### PR DESCRIPTION
**Pull request by fredykoci**
Fixes: https://github.com/It4innovations/hyperqueue/issues/37

### CODE REVIEW

File: src/bin/hq.rs [hunk 1]
```diff
@@ -201,15 +219,38 @@ async fn command_worker_stop(
...
-        .map_err(|e| e.into())
+    let mut workers = get_worker_list(&mut connection, &gsettings)
+        .await.unwrap();
+    let mut print_workers = Vec::new();
+
+    for worker in workers{
+        if worker_state(&worker,&_opts){
+            print_workers.push(worker)
+        }
+    }
```
Comment (Kobzol) [inline src/bin/hq.rs:249]:
```rust
let workers_to_print: Vec<_> = workers
    .iter()
    .filter(|worker| worker_state(worker, &opts)
    .collect();
```

File: src/bin/hq.rs [hunk 2]
```diff
@@ -201,15 +219,38 @@ async fn command_worker_stop(
         .await
         .map_err(|e| e.into())
 }
-
+fn worker_state(worker:&WorkerInfo,option:&WorkerListOpts) -> bool{
```
Comment (Kobzol) [inline src/bin/hq.rs:222]: I would rename this to something like `should_be_printed`.

**Review by Kobzol — CHANGES_REQUESTED** [review]
With this implementation, if the user passes neither `running` nor `offline`, no workers will be shown.
Maybe it would be better to only implement a single flag (`--online`) with the following behaviour:
1) If nothing is passed, all workers will be displayed
2) If `--online` is passed, only online workers will be displayed

File: src/bin/hq.rs [hunk 3]
```diff
@@ -201,15 +219,43 @@ async fn command_worker_stop(
...
 async fn command_worker_list(
     gsettings: GlobalSettings,
     _opts: WorkerListOpts,
 ) -> anyhow::Result<()> {
     let mut connection = get_client_connection(&gsettings.server_directory()).await?;
-    get_worker_list(&mut connection, &gsettings)
-        .await
-        .map_err(|e| e.into())
+    let mut workers = get_worker_list(&mut connection, &gsettings)
+        .await.unwrap();
```
Comment (spirali) [inline src/bin/hq.rs:242]: Do not use unwrap but propagate the error (via ``?`` operator)

File: src/client/commands/worker.rs [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
 use crate::transfer::connection::ClientConnection;
-use crate::transfer::messages::{FromClientMessage, StopWorkerMessage, ToClientMessage};
+use crate::transfer::messages::{FromClientMessage, StopWorkerMessage, ToClientMessage, WorkerInfo};
 use crate::WorkerId;
 
 pub async fn get_worker_list(
     connection: &mut ClientConnection,
     gsettings: &GlobalSettings,
-) -> crate::Result<()> {
+) -> crate::Result<(Vec<WorkerInfo>)> {
```
Comment (spirali) [inline src/client/commands/worker.rs:10]: Superfluous parenthesis

**Review by spirali — CHANGES_REQUESTED** [review]

**Comment by spirali** [comment]
Please include also Python integration tests in your code (directory "tests").

**Comment by spirali** [comment]
I have rebased and squashed your commits as c9e6523d6806a36ef461db5efa1c5f5d23db719b.
I have also added a small refactoring 7144980369a5a747b029ff5710e304dab59e719e, if you have any question to my changes, do not hesitate to ask.

Thank you for your work.

**Pull request closed without merging**